subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/CompleteDirectorySnapshot.java

```diff
@@ -65,6 +67,15 @@ public void accept(FileSystemSnapshotVisitor visitor) {
         visitor.postVisitDirectory(this);
     }
 
+    @Override
+    public void accept(KnownNodeVisitor visitor) {
+        visitor.visitKnownNode(getPathToParent(), this);
+        children.forEach(child -> child.accept((pathToParent, node) -> {
```

Comment:
Why are we visiting the children here? Shouldn't it be enough to watch the current directory?

Reply:
See other comment.